### PR TITLE
feat(GAME-018): scenario select screen + progression persistence

### PR DESCRIPTION
**Pull request by cgruber**
## Prerequisites
- [x] Parent PR merged: #74 (GAME-017 evaluation phase)
- [x] Parent PR merged: #75 (GAME-017 close ticket)

## Summary
- Implements GAME-018: scenario select screen + localStorage completion persistence
- New players see the intro screen directly on first load
- Returning players (any completed scenario in localStorage) land on scenario select
- Passing the evaluation marks the scenario complete, saves progress, enables "Next Scenario" → select screen
- Progress uses pure serialization functions (testable without mocking localStorage)
- 18 unit tests for progress.ts serialization/mutation; 5 Playwright e2e tests for progression flows

## Changes
- `game/web/index.html`: `#intro-screen` starts hidden (class="hidden"); JS shows it on startup routing
- `game/web/src/main.ts`:
  - Scenario manifest (hardcoded for v1 with 1 scenario)
  - `startScenarioIntro()` with AbortController for clean re-entry on "Play Again"
  - `renderScenarioCards()` / `showScenarioSelect()` for scenario select screen
  - Progress-aware startup routing: returning player → select, new player → intro
  - `showResultScreen()` saves progress to localStorage on pass
  - "Next Scenario" button routes to scenario select (re-renders cards with updated state)
- `game/web/src/model/progress.ts`: already on main (pure functions + localStorage I/O)
- `game/web/src/model/progress_test.ts`: 18 unit tests (new file)
- `game/web/src/model/BUILD.bazel`: added `progress_test_lib`, `progress_test`
- `game/web/e2e/sprint3.spec.ts`: 5 GAME-018 e2e tests added

## Validation performed
- [x] `bazel build //web:app_typecheck_lib` passes
- [x] `bazel test //web/src/model:progress_test` passes (18 tests)
- [x] `bazel test //web/src/simulation:evaluate_test //web/src/model:loader_test` cached pass

## Tickets addressed
- see #76

### CODE REVIEW

File: game/web/e2e/sprint3.spec.ts [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
- *   10. "Keep Drawing" button hides the result screen
+ *   7. Submit button remains in DOM after painting precincts
+ *   8. Clicking submit shows result screen with criteria
+ *   9. "Keep Drawing" button hides the result screen
+ *
+ * GAME-018: Progression:
+ *   10. Scenario select screen is shown for returning players (localStorage has completion data)
+ *   11. Scenario card shows "Completed" status after completing a scenario
+ *   12. Page reload restores completion state from localStorage
+ *   13. New player (no localStorage) sees intro, not scenario select
```
Comment (cgruber) [inline game/web/e2e/sprint3.spec.ts]: Nit: the GAME-018 doc comment block lists four tests (numbered 10–13), but five are actually implemented. The "Play Again from select screen shows intro then editor" test (line ~213) is missing from the index at the top of the file. Please add it as item 11 (and renumber the existing 11–13 to 12–14, or keep the numbering flat).

File: game/web/src/main.ts [hunk 2]
```diff
@@ -381,9 +389,53 @@ if (wasmEl !== null) {
...
 	});
 
-	// "Next Scenario" is a placeholder until GAME-018 adds the select screen
+	// ── Scenario select screen (GAME-018) ─────────────────────────────────────
+	function renderScenarioCards() {
+		if (!scenarioCardsEl) return;
+		scenarioCardsEl.innerHTML = "";
+		SCENARIO_MANIFEST.forEach((entry, i) => {
+			const completed = isCompleted(progress, entry.id);
+			const unlocked = i === 0 || isCompleted(progress, SCENARIO_MANIFEST[i - 1]!.id);
```
Comment (cgruber) [inline game/web/src/main.ts]: Suggestion: with `noUncheckedIndexedAccess` enabled in tsconfig, `SCENARIO_MANIFEST[i - 1]` has type `{ id: string; title: string } | undefined`. The non-null assertion `!` bypasses the exact safety net the project opted into.

The logic is correct at runtime — when `i === 0` the left side of `||` short-circuits so `SCENARIO_MANIFEST[-1]!` is never evaluated, and when `i > 0` the element is always defined — but the assertion suppresses the compiler warning unnecessarily.

A type-safe alternative that degrades correctly:

```ts
const unlocked = i === 0 || isCompleted(progress, SCENARIO_MANIFEST[i - 1]?.id ?? "");
```

An empty-string id will never match any real scenario id, so `isCompleted` returns false → the scenario stays locked. No behaviour change, no assertion.

File: game/web/src/model/progress_test.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	if (condition) {
+		console.log(`ok ${total} - ${description}`);
+		passed++;
+	} else {
+		console.log(`not ok ${total} - ${description}`);
+		failed++;
+	}
+}
+
+function assertEqual<T>(description: string, actual: T, expected: T): void {
```
Comment (cgruber) [inline game/web/src/model/progress_test.ts]: Nit: `assertEqual` is defined here but never called anywhere in the file — `assertDeepEqual` is used for all structural checks instead. `noUnusedLocals` isn't set so this compiles fine, but it's dead code. Either add a test that uses it (e.g., testing that `serializeProgress` returns a specific exact string) or remove the function.

**Comment by cgruber** [comment]
LGTM with nits. The GAME-018 implementation is well-structured:

- AbortController refactor correctly prevents duplicate intro listeners on replay — a genuine improvement over the manual `escHandler` cleanup.
- `progress = markCompleted(...)` re-assignment is valid (`let` at correct scope).
- `seedProgress()` via `addInitScript` is the right Playwright pattern; it re-runs on `page.reload()` as the reload test relies on.
- The `i === 0` short-circuit makes the `SCENARIO_MANIFEST[i - 1]!.id` safe at runtime (see inline suggestion for a cleaner type-safe form).
- 18 unit tests in `progress_test.ts` cover all four exported functions including edge cases (malformed JSON, null JSON, non-string array items, idempotency, immutability).
- 5 e2e tests cover the full progression flow (new player, returning player, card status, reload persistence, Play Again path).

Three inline nits posted — none are blocking.

File: game/web/e2e/sprint3.spec.ts (hunk 1 above)
Comment (cgruber) [reply game/web/e2e/sprint3.spec.ts]: Fixed. Updated the GAME-018 section in the doc comment to list all 5 tests (10–14), including "Play Again from select screen shows intro then editor".

File: game/web/src/main.ts (hunk 2 above)
Comment (cgruber) [reply game/web/src/main.ts]: Fixed. Changed to `SCENARIO_MANIFEST[i - 1]?.id ?? ""`. The empty string is safe: `isCompleted(progress, "")` always returns false since no scenario has an empty-string id, so the unlock chain behaves identically while satisfying `noUncheckedIndexedAccess`.

File: game/web/src/model/progress_test.ts (hunk 3 above)
Comment (cgruber) [reply game/web/src/model/progress_test.ts]: Fixed. Removed the unused `assertEqual` function; only `assertDeepEqual` and `assert` are needed for the progress test cases.
